lib/bricks-net-api-types.js

```diff
@@ -0,0 +1,1176 @@
+		client.request_method_ = "GET";
+		client.request_url_ = request.url;
+		if (request.custom_user_agent) {
+			client.request_user_agent_ = request.custom_user_agent;
+		}
+	}
+	else if (request instanceof POST) {
+		client.request_method_ = "POST";
+		client.request_url_ = request.url;
+		if (request.custom_user_agent) {
+			client.request_user_agent_ = request.custom_user_agent;
+		}
+		client.request_has_body_ = request.has_body;
+		if (request.has_body) {
+			client.request_body_contents_ = request.body;
+			client.request_body_content_type_ = request.content_type;
+		}
+	}
+	else if (request instanceof POSTFromFile) {
+		client.request_method_ = "POST";
+		client.request_url_ = request.url;
+		if (request.custom_user_agent) {
+			client.request_user_agent_ = request.custom_user_agent;
+		}
+		client.request_has_body_ = true;
+		client.request_body_contents_ = fs.readFileSync(request.file_name);  // Can throw.
+		client.request_body_content_type_ = request.content_type;
+	}
+	else if (request instanceof KeepResponseInMemory) {
+		// The `request` here is actually `response_params`.
+	}
+	else if (request instanceof SaveResponseToFile) {
+		// The `request` here is actually `response_params`.
+		if (!request.file_name || typeof request.file_name !== 'string') {
+			throw new Error('Property "file_name" must be a non-empty string on SaveResponseToFile.');
+		}
+	}
+	else {
+		throw new Error('Expected one of: GET, POST, POSTFromFile, KeepResponseInMemory, SaveResponseToFile.');
+	}
+}
+
+function ParseOutput(request_params, response_params, response, output) {
+	if (!(output instanceof HTTPResponse)) {
+		throw new Error('Argument "output" must be HTTPResponse, ' + (typeof output) + ' given.');
+	}
+	
+	when(response).then(function () {
+		if (!request_params.allow_redirects && request_params.url != response.response_url_after_redirects_) {
+			BRICKS_THROW(HTTPRedirectNotAllowedException());
+		}
+		output.url = response.response_url_after_redirects_;
+		output.code = response.response_code_;
+		
+		if (output instanceof HTTPResponseWithBuffer) {
+			var http_request = response.HTTPRequest();
+			output.body = http_request.HasBody() ? http_request.Body() : "";
+		}
+		else if (output instanceof HTTPResponseWithResultingFileName) {
+			var http_request = response.HTTPRequest();
+			// TODO(sompylasar): Make this an async writeFile and return its promise.
+			fs.writeFileSync(response_params.file_name, http_request.HasBody() ? http_request.Body() : "");
+			output.body_file_name = response_params.file_name;
+		}
+	}).done(function () {
+		output.resolver.resolve(output);
+	}, function (err) {
+		output.resolver.reject(err);
+	});
+}
+
+
+function HTTP() {
+	var _this = this;
+	var retval;
+	
+	cppArguments.assert('HTTP', [
+		[
+			cppArguments.assertion(function (value) {
+				return (value instanceof HTTPRequestBase);
+			}, 'const T_REQUEST_PARAMS&', 'request_params'),
+			cppArguments.assertion(function (value) {
+				return (
```

Comment:
It barely fits one line of code, but surely not one line of the reader's mind, I think. :)